tests/compiler/test_plugins.py

```diff
@@ -431,14 +436,53 @@ def test_context_and_defaults(snippetcompiler: "SnippetCompilationTest") -> None
     compiler.do_compile()
 
 
-def test_native_types(snippetcompiler: "SnippetCompilationTest") -> None:
+def test_inferred_signatures_logging(snippetcompiler: "SnippetCompilationTest", caplog) -> None:
     """
-    test the use of python types
+    Test that the signature (using inferred Inmanta types)
+    for each plugin is correctly logged
     """
-    snippetcompiler.setup_for_snippet(
-        """
+    with caplog.at_level(logging.DEBUG):
+
+        snippetcompiler.setup_for_snippet(
+            """
 import plugin_native_types
+            """
+        )
+        compiler.do_compile()
 
+        expected_signatures = [
+            "get_from_dict(value: dict[string], key: string) -> string?",
```

Comment:
`dict[string]` is a tough one. I think it could be confusing, because it is in contrast with Python's dict semantics. But `dict[string, string]` could also be confusing because we really only have on type of dicts. Overall I think I prefer it though. Wdyt?

Reply:
it is a bit messy, as the native dsl types don't have typed dicts, so I find it hard as well

Reply:
@wouterdb CC